templates/types/streaming/nextjs/app/components/ui/chat/chat-sources.tsx

```diff
@@ -23,55 +66,66 @@ export function ChatSources({ data }: { data: SourceData }) {
     <div className="space-x-2 text-sm">
       <span className="font-semibold">Sources:</span>
       <div className="inline-flex gap-1 items-center">
-        {sources.map((node: SourceNode, index: number) => (
-          <div key={node.id}>
-            <HoverCard>
-              <HoverCardTrigger>
-                <div className="text-xs w-5 h-5 rounded-full bg-gray-100 mb-2 flex items-center justify-center hover:text-white hover:bg-primary hover:cursor-pointer">
-                  {index + 1}
-                </div>
-              </HoverCardTrigger>
-              <HoverCardContent>
-                <NodeInfo node={node} />
-              </HoverCardContent>
-            </HoverCard>
-          </div>
-        ))}
+        {sources.map((node: SourceNode, index: number) => {
+          const nodeInfo = getNodeInfo(node);
+          if (nodeInfo?.path.endsWith(".pdf")) {
+            return (
+              <PdfDialog
+                key={node.id}
+                documentId={node.id}
+                filePath={nodeInfo.path}
+                trigger={<SourceNumberButton index={index} />}
+              />
+            );
+          }
+          return (
+            <div key={node.id}>
+              <HoverCard>
+                <HoverCardTrigger>
+                  <SourceNumberButton index={index} />
+                </HoverCardTrigger>
+                <HoverCardContent className="w-[320px]">
+                  <NodeInfo node={node} />
+                </HoverCardContent>
+              </HoverCard>
+            </div>
+          );
+        })}
       </div>
     </div>
   );
 }
 
 function NodeInfo({ node }: { node: SourceNode }) {
   const { isCopied, copyToClipboard } = useCopyToClipboard({ timeout: 1000 });
+  const nodeInfo = getNodeInfo(node);
 
-  if (typeof node.metadata["URL"] === "string") {
+  if (nodeInfo?.type === NODE_TYPE.URL) {
     // this is a node generated by the web loader, it contains an external URL
     // add a link to view this URL
     return (
       <a
         className="space-x-2 flex items-center my-2 hover:text-blue-900"
```

Comment:
as we're serving now local files, can we also use a link for them? (same as for URLs)